Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/InMemorySpawnableAssetContainer.cpp

```diff
@@ -213,63 +212,91 @@ namespace AzToolsFramework::Prefab::PrefabConversionUtils
+                    if (loadBehavior == AZ::Data::AssetLoadBehavior::NoLoad)
+                    {
+                        return false;
+                    }
+
+                    AZ::Data::AssetId assetId = asset->GetId();
+                    AZ::Data::AssetType assetType = asset->GetType();
+
+                    // Always queue load as the next step will stop loading for all PreLoad assets
+                    *asset = AZ::Data::AssetManager::Instance().GetAsset(assetId, assetType, AZ::Data::AssetLoadBehavior::QueueLoad);
+
+                    if (!asset->GetId().IsValid())
+                    {
+                        AZ_Error("Prefab", false, "Invalid asset found referenced in scene while entering game mode");
+                        return false;
+                    }
+
+                    if (loadBehavior == AZ::Data::AssetLoadBehavior::PreLoad)
+                    {
+                        // Only assets that are preloaded need to be waited on.
+                        blockingAssets.push_back(asset);
+                    }
+                    return false;
+                }
+                return true;
+            };
+
+            AZ::SerializeContext::EnumerateInstanceCallContext enumerationContext(
+                callback, nullptr, sc, AZ::SerializeContext::ENUM_ACCESS_FOR_READ, nullptr);
+            sc->EnumerateInstance(&enumerationContext, asset.GetData(), asset.GetType(), nullptr, nullptr);
         }
 
         // For all Preload assets we block until they're ready
         // We do this as a separate pass so that we don't interrupt queuing up all other asset loads
-        for (AZ::Data::Asset<AZ::Data::AssetData>& asset : referencedAssets)
+        for (AZ::Data::Asset<AZ::Data::AssetData>* asset : blockingAssets)
         {
-            if (!asset.GetId().IsValid())
+            asset->BlockUntilLoadComplete();
+            
+            if (asset->IsError())
             {
-                AZ_Error("Prefab", false, "Invalid asset found referenced in scene while entering game mode");
-                continue;
-            }
-
-            const AZ::Data::AssetLoadBehavior loadBehavior = asset.GetAutoLoadBehavior();
+                AZ_Error(
+                    "Prefab", false, "Asset with id %s failed to preload while entering game mode",
+                    asset->GetId().ToString<AZStd::string>().c_str());
 
-            if (loadBehavior != AZ::Data::AssetLoadBehavior::PreLoad)
-            {
                 continue;
             }
 
-            asset.BlockUntilLoadComplete();
-
-            if (asset.IsError())
-            {
-                AZ_Error("Prefab", false, "Asset with id %s failed to preload while entering game mode",
-                    asset.GetId().ToString<AZStd::string>().c_str());
-
-                continue;
-            }
+            // Reset the load behavior back to preload because the async load will have caused the behavior to be set to queued. Some assets
```

Comment:
Not for this PR, but it feels like a utility for "blocking load N assets" that handles these nuances for you should exist.

Reply:
This is probably one of the few places that can make practical use of it as most asset loading is triggered from serialization. On the upside, Asset Streaming does the above trick on a larger scale for any asset dependencies, which is why it's so important dependencies get registered correctly in the AP.